### PR TITLE
[SAH-39]: Sort Products in Supplier Page

### LINKED ISSUE #78: [SAH-39]: Sort Products in Supplier Page

### PR DESCRIPTION
**Pull request by Maiz27**
### Tasks

- [x] Allow sorting products by name (ASC & DESC). 
- [x] Allow sorting products by quantity (ASC & DESC). 
- [x] Allow sorting products by price (ASC & DESC).

### CODE REVIEW

File: apps/agent/src/Suppliers/SupplierProductList.tsx [hunk 1]
```diff
@@ -42,8 +44,45 @@ const SupplierProductList = ({ productsCount }: props) => {
     });
   };
 
+  const sortOptions = [
```
Comment (Emmanuel-Melon) [inline apps/agent/src/Suppliers/SupplierProductList.tsx:50]: I think it'd be a great idea to add sorting and filtering to `<ListHeader />` for reusability across different pages.

File: apps/agent/src/Suppliers/SupplierProductList.tsx [hunk 2]
```diff
@@ -42,8 +44,45 @@ const SupplierProductList = ({ productsCount }: props) => {
...
+    "Quantity ASC",
+    "Quantity DESC",
+    "Price ASC",
+    "Price DESC",
+  ];
+
   return (
     <div className="space-y-4">
+      <div className="flex justify-end items-center gap-4">
+        <select className="select select-bordered select-sm max-w-xs">
```
Comment (Emmanuel-Melon) [inline apps/agent/src/Suppliers/SupplierProductList.tsx]: How about using a custom select component?

File: apps/agent/src/Suppliers/SupplierProductList.tsx (hunk 1 above)
Comment (Maiz27) [reply apps/agent/src/Suppliers/SupplierProductList.tsx:50]: Good idea, I will refactor `<ListHeader />` accordingly.

**Review by Emmanuel-Melon — APPROVED** [review]
LGTM

**Pull request merged**